### PR TITLE
chore: fix builds/tests/eslint

### PR DESCRIPTION
**Pull request by balazsorban44**
ESLint has not been working for me for a while now, I finally had a look at it to fix it. Builds also started failing locally, because of some missing types, which should be fixed now as well.

I also turned on linting on test files.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const path = require('path')
```
Comment (ThangHuuVu) [inline .eslintrc.js:1]: I moved eslint config to here to use the `path.resolve` in `parserOptions.project`

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "include": [
+    ".eslintrc.js",
```
Comment (ThangHuuVu) [inline tsconfig.json:3]: This file is included to run lint on `eslintrc.js`

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.eslintrc.js
```
Comment (balazsorban44) [inline .eslintignore:1]: Can we make this not an extra file? :sweat_smile:

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+module.exports = {
+  root: true,
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    project: [
+      path.resolve(__dirname, './packages/**/tsconfig.eslint.json'),
```
Comment (balazsorban44) [inline .eslintrc.js:8]: Would be happy if this did not need to move out of `package.json`, but I see you are using `path`. I won't have strong feelings about it if there is no workaround. :thinking:

Comment (ThangHuuVu) [reply .eslintrc.js]: I moved it inside `pacakge.json` again! I thought missing `path` and `__dirname` was the problem but it turned out weren't

File: .eslintignore (hunk 3 above)
Comment (ThangHuuVu) [reply .eslintignore]: Removed because eslintConfig is restored in `package.json`

**Review by ndom91 — APPROVED** [review]
Tests pass.. LGTM 😊👍

**Comment by ThangHuuVu** [comment]
@ndom91 great! @balazsorban44 and Nico, could you check if it works on your local machine? 🙌

**Comment by ndom91** [comment]
> @ndom91 great! @balazsorban44 and Nico, could you check if it works on your local machine? 🙌

Yeah, can check later this evening when I'm back home 👍

**Comment by ThangHuuVu** [comment]
In your IDE, make sure to add Prettier `configPath` for auto-save:
```
{
  "prettier.configPath": "package.json"
}
```
Merging this now, I pulled a fresh repo and it still works 🤞

**Review by ThangHuuVu — APPROVED** [review]

**Pull request merged**